### PR TITLE
MGDAPI-5786

### PR DESCRIPTION
**Pull request by MStokluska**
# Issue link
https://issues.redhat.com/browse/MGDAPI-5786

# What
Changes to support 3scale 0.11.6

# Verification steps
## Fresh install
- Include the changes coming from this PR to https://github.com/trepel/integreatly-operator/tree/3scale-next-0.11.6-mas
- Add integreatly-delorean-auth.json to main folder of integreatly operator
- Run make cluster/prepare/local LOCAL=false
- Run make code/run
- Confirm installation has been completed successfully and that API Manager isn't continuously updated

Upgrade:
- Include the changes coming from this PR to https://github.com/trepel/integreatly-operator/tree/3scale-next-0.11.6-mas
- Run `make create/olm/bundle OLM_TYPE=managed-api-service UPGRADE=false BUNDLE_VERSIONS=1.38.0,1.37.0 ORG=<YOUR QUAY ORG>`
- Prepare the cluster without the integreatly delorean secret (skip the mirror mapping)
- Install 1.37.0
- Once installed - apply the mirror mapping file to the cluster via https://github.com/integr8ly/integreatly-operator/blob/master/Makefile#L410
- Apply 1.38 upgrade
- Confirm 1.38 upgrades and that API Manager isn't continuously updated

### CODE REVIEW

File: pkg/products/threescale/reconciler.go [hunk 1]
```diff
@@ -3062,121 +3066,6 @@ func (r *Reconciler) deleteConsoleLink(ctx context.Context, serverClient k8sclie
...
-		}
-
-		podPriorityMutation := resources.MutatePodPriority(r.installation.Spec.PriorityClassName)
-
-		phase, err := resources.UpdatePodTemplateIfExists(
-			ctx,
-			serverClient,
-			resources.SelectFromDeploymentConfig,
-			resources.AllMutationsOf(
-				resources.MutateZoneTopologySpreadConstraints("app"),
```
Comment (KevFan) [inline pkg/products/threescale/reconciler.go:3082]: We probably still need `resources.MutateZoneTopologySpreadConstraints("app")` for zone topology spread constaints for multi az unless we can also define this via the ApiManager CR on 0.11.6-mas 🤔

Comment (MStokluska) [reply pkg/products/threescale/reconciler.go:3082]: yes. This is still in progress

File: pkg/products/threescale/reconciler_test.go [hunk 2]
```diff
@@ -592,6 +592,19 @@ func TestReconciler_reconcileComponents(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	nodes := &corev1.NodeList{
```
Comment (MStokluska) [inline pkg/products/threescale/reconciler_test.go:595]: this isn't necessary yet - but will be once topology constraints go in.

**Comment by Patryk-Stefanski** [comment]
upgrade scenario was successful,  /lgtm

**Pull request merged**